azure-kusto-ingest/azure/kusto/ingest/_kusto_ingest_client.py

```diff
@@ -27,69 +29,124 @@ def __init__(self, kcsb):
+
+        df.to_csv(temp_file_path, index=False, encoding='utf-8', header=False, compression='gzip')
+
+        fd = FileDescriptor(temp_file_path)
+
+        blob_name = "{db}__{table}__{guid}__{file}".format(
+            db=ingestion_properties.database,
+            table=ingestion_properties.table,
+            guid=uuid.uuid4(),
+            file=file_name,
+        )
+
+        containers = self._resource_manager.get_containers()
+        container_details = random.choice(containers)
+        storage_client = CloudStorageAccount(
+            container_details.storage_account_name, sas_token=container_details.sas
+        )
+        blob_service = storage_client.create_block_blob_service()
+
+        blob_service.create_blob_from_path(
+            container_name=container_details.object_name,
+            blob_name=blob_name,
+            file_path=temp_file_path            
+        )                
+
+        url = blob_service.make_blob_url(
+            container_details.object_name,
+            blob_name,
+            sas_token=container_details.sas,
+        )
+
+        blobs.append(BlobDescriptor(url, fd.size))
+
+        self.ingest_from_blobs(
+            blobs,
+            ingestion_properties=ingestion_properties
+        )
+
+        fd.delete_files()
+        os.unlink(temp_file_path)
+
+    def ingest_from_files(
+        self, files, ingestion_properties
+    ):
         """Enqueuing an ingest command from local files.
         :param files: List of FileDescriptor or file paths. The list of files to be ingested.
-        :param bool delete_sources_on_success: After a successful ingest,
-            whether to delete the origin files.
         :param azure.kusto.ingest.IngestionProperties ingestion_properties: Ingestion properties.
         """
         blobs = list()
         file_descriptors = list()
+        containers = self._resource_manager.get_containers()
+
         for file in files:
             if isinstance(file, FileDescriptor):
                 descriptor = file
             else:
-                descriptor = FileDescriptor(file, deleteSourcesOnSuccess=delete_sources_on_success)
+                descriptor = FileDescriptor(file)
+
             file_descriptors.append(descriptor)
-            blob_name = (
-                ingestion_properties.database
-                + "__"
-                + ingestion_properties.table
-                + "__"
-                + text_type(uuid.uuid4())
-                + "__"
-                + descriptor.stream_name
+            blob_name = "{db}__{table}__{guid}__{file}".format(
+                db=ingestion_properties.database,
+                table=ingestion_properties.table,
+                guid=uuid.uuid4(),
+                file=descriptor.stream_name,
             )
-            containers = self._resource_manager.get_containers()
+
+            # TODO: maybe should only pick one for entire ingestion?
```

Comment:
No. We choose randomly for each blob. It costs nothing and it reduces throttling. Please remove this TODO